composer/callbacks/checkpoint_saver.py

```diff
@@ -358,60 +347,50 @@ def _save_checkpoint(self, state: State, logger: Logger, log_level: LogLevel):
 
         is_deepspeed = is_model_deepspeed(state.model)
 
-        if is_deepspeed and '{rank}' not in self.filename.filename:
-            raise ValueError(f'Save filename {self.filename.filename} must have {{rank}} for deepspeed.')
-
-        # save the checkpoint to the filename
-        filename = self.filename.format(state, is_deepspeed)
-
         saved_path = checkpoint.save_checkpoint(
             state=state,
-            filename=filename,
+            filename=self.file_path,
             weights_only=self.weights_only,
         )
 
+        checkpoint_file_path = format_name_with_dist_and_time(self.checkpoint_file_path, state.run_name, state.timestamp)
+        checkpoint_file_path += ('.tar' if is_deepspeed and not is_tar(checkpoint_file_path) else '')
+
         if not saved_path:  # not all ranks save
             return
 
         if self.latest_filename is not None:
-            symlink = self.latest_filename.format(state, is_deepspeed)
-            os.makedirs(os.path.dirname(symlink), exist_ok=True)
+            symlink_name = format_name_with_dist_and_time(self.latest_filename, state.run_name, state.timestamp)
+            symlink_name += ('.tar' if is_deepspeed and not is_tar(self.latest_filename) else '')
+            symlink_path = os.path.join(self.folder, symlink_name)
+            os.makedirs(os.path.dirname(symlink_path), exist_ok=True)
             try:
-                os.remove(symlink)
+                os.remove(symlink_path)
             except FileNotFoundError:
                 pass
-            os.symlink(os.path.relpath(filename, os.path.dirname(symlink)), symlink)
+            os.symlink(os.path.relpath(checkpoint_file_path, os.path.dirname(symlink_path)), symlink_path)
 
-        # if artifact name provided, upload the checkpoint
-        if self.artifact_name is not None:
-            artifact_name = self.artifact_name.format(
-                state,
-                is_deepspeed,
-            ).lstrip('/')
 
+      
+        if self.backend is not None:
             logger.file_artifact(log_level=log_level,
-                                 artifact_name=artifact_name,
-                                 file_path=filename,
+                                 artifact_name=checkpoint_file_path,
```

Comment:
Does this remove the ability to name the checkpoint file differently locally and on the cloud? I think we should probably preserve that. Also the old default for the artifact name is probably a better default than the current default for the filename